src/pages/test-report/cat-be/components/reverse-diagram-modal/reverse-diagram-modal.ts

```diff
@@ -0,0 +1,109 @@
+import { Component, Input, OnInit } from '@angular/core';
+import { Store, select } from '@ngrx/store';
+import { Observable } from 'rxjs/Observable';
+import { merge } from 'rxjs/observable/merge';
+import { Subscription } from 'rxjs/Subscription';
+import { map } from 'rxjs/operators';
+import { IonicPage, NavParams } from 'ionic-angular';
+import { StoreModel } from '../../../../../shared/models/store.model';
+import { getTests } from '../../../../../modules/tests/tests.reducer';
+import { getVehicleDetails } from '../../../../../modules/tests/vehicle-details/vehicle-details.cat-be.reducer';
+import {
+  getVehicleLength,
+  getVehicleWidth,
+} from '../../../../../modules/tests/vehicle-details/vehicle-details.cat-be.selector';
+import { getCurrentTest } from '../../../../../modules/tests/tests.selector';
+
+interface ReverseDiagramPageState {
+  vehicleLength$: Observable<number>;
+  vehicleWidth$: Observable<number>;
+}
+
+@IonicPage()
+@Component({
+  selector: 'reverse-diagram-modal',
+  templateUrl: 'reverse-diagram-modal.html',
+})
+
+export class ReverseDiagramCatBEPage implements OnInit {
+  @Input()
+  vehicleLength: number;
+
+  @Input()
+  vehicleWidth: number;
+
+  onClose: Function;
```

Comment:
We can be a bit more specific here with something like -
`type OnCloseFunc = () => void;`
Then using it like -
`onClose: OnCloseFunc;`